src/screens/ImportWalletQRCodeScreen.tsx

```diff
@@ -65,220 +69,396 @@ export default class ImportWalletQRCodeScreen extends React.Component<Props, Sta
+
+    // if (newWallet.getAddress() === false && newLegacyWallet.getAddress() === false) {
+    //   Alert.alert(loc.wallets.scanQrWif.bad_wif);
+    //   this.setState({ isLoading: false });
+    //   return;
+    // }
+
+    // if (newWallet.getAddress() === false && newLegacyWallet.getAddress() !== false) {
+    //   // case - WIF is valid, just has uncompressed pubkey
+    //   newLegacyWallet.setLabel(loc.wallets.scanQrWif.imported_legacy);
+    //   BlueApp.wallets.push(newLegacyWallet);
+    //   Alert.alert(
+    //     loc.wallets.scanQrWif.imported_wif +
+    //       event.data +
+    //       loc.wallets.scanQrWif.with_address +
+    //       newLegacyWallet.getAddress(),
+    //   );
+    //   await newLegacyWallet.fetchBalance();
+    //   await newLegacyWallet.fetchTransactions();
+    //   await BlueApp.saveToDisk();
+    //   this.props.navigation.popToTop();
+    //   setTimeout(() => EV(EV.enum.WALLETS_COUNT_CHANGED), 500);
+    //   return;
+    // }
+
+    // this.setState({ isLoading: true });
+    // await newLegacyWallet.fetchBalance();
+    // console.log('newLegacyWallet == ', newLegacyWallet.getBalance());
+
+    // if (newLegacyWallet.getBalance()) {
+    //   newLegacyWallet.setLabel(loc.wallets.scanQrWif.imported_legacy);
+    //   BlueApp.wallets.push(newLegacyWallet);
+    //   Alert.alert(
+    //     loc.wallets.scanQrWif.imported_wif +
+    //       event.data +
+    //       loc.wallets.scanQrWif.with_address +
+    //       newLegacyWallet.getAddress(),
+    //   );
+    //   await newLegacyWallet.fetchTransactions();
+    // } else {
+    //   await newWallet.fetchBalance();
+    //   await newWallet.fetchTransactions();
+    //   newWallet.setLabel(loc.wallets.scanQrWif.imported_segwit);
+    //   BlueApp.wallets.push(newWallet);
+    //   Alert.alert(
+    //     loc.wallets.scanQrWif.imported_wif + event.data + loc.wallets.scanQrWif.with_address + newWallet.getAddress(),
+    //   );
+    // }
+    // await BlueApp.saveToDisk();
+    // this.props.navigation.popToTop();
+    // setTimeout(() => EV(EV.enum.WALLETS_COUNT_CHANGED), 500);
+  };
+
+  saveWallet = async (w: any) => {
+    if (BlueApp.getWallets().some((wallet: Wallet) => wallet.getSecret() === w.secret)) {
+      Alert.alert('walletInUseValidationError');
+    } else {
+      w.setLabel(i18n.wallets.import.imported + ' ' + w.typeReadable);
+      BlueApp.wallets.push(w);
+      await BlueApp.saveToDisk();
+      EV(EV.enum.WALLETS_COUNT_CHANGED);
+      this.props.navigation.popToTop();
+
+      this.showSuccessImportMessageScreen();
+      // this.props.navigation.dismiss();
     }
+  };
 
-    for (const w of BlueApp.wallets) {
-      if (w.getSecret() === event.data) {
-        // lookig for duplicates
-        this.setState({ isLoading: false });
-        return Alert.alert(loc.wallets.scanQrWif.wallet_already_exists); // duplicate, not adding
+  importMnemonic = async (text: string) => {
+    console.log('importMnemonic');
```

Comment:
Would be great to gradually remove all console logs as they hurt performance in production

Reply:
Agree, console.logs should be treated as errors by eslint

Reply:
https://trello.com/c/qQQ43w0P/218-add-eslint-rule-for-consolelogs I've created a task for that